File: Assignment 2/Codes/find_coordinates.c
#include <stdio.h>
#include <math.h>

float y_coords(float x)  {
return x*x;
}

float x_coords(float a, float b, float c)  {

    static int counter = 0;
    float disc = b*b - 4*a*c;

    if(counter == 0) {
        counter++;
        return (-b - sqrt(disc))/2*a;
    } else return (-b + sqrt(disc))/2*a;
    
}

int main()  {

    float a = 1, b = 1, c = -2;               //solving ax^2 + bx + c = 0
    float x1, x2; 

    x1 = x_coords(a,b,c);
    x2 = x_coords(a,b,c);

    float y1 = y_coords(x1), y2 = y_coords(x2);

    printf("The points of intersection are (%.0f,%.0f) and (%.0f,%.0f)\n",x1,y1,x2,y2);

return 0;
}

File: Assignment 2/Codes/find_coordinates.py
import math

def quadratic_solve(a,b,c):
    disc = b**2 - 4*a*c

    if disc < 0:
        print("The curves do not intersect")
    elif disc == 0:
        x = -b/2*a
        y = x**2
        print("The curves intersect at a unique point ({},{})" .format(x,y))
    elif disc > 0:
        x1 = (-b - math.sqrt(disc))/2*a
        x2 = (-b + math.sqrt(disc))/2*a
        y1 = x1**2
        y2 = x2**2
        print("The curves intersect at two distinct points ({},{}) and ({},{})" .format(x1,y1,x2,y2))

# the quadratic is x**2 + x - 2 = 0
quadratic_solve(1,1,-2)